posthog/api/__init__.py

```diff
@@ -42,6 +42,7 @@ def api_not_found(request):
 # Legacy endpoints (to be removed eventually)
 router.register(r"annotation", annotation.AnnotationsViewSet)
 router.register(r"feature_flag", feature_flag.FeatureFlagViewSet)
+router.register(r"feature_flag_override", feature_flag.FeatureFlagOverrideViewset)
```

Comment:
We really really should not add new endpoints like this (notice the "Legacy endpoints" comment), because they are the crux of a problem we plan to solve soon – being completely unable to use different projects at the same time (https://github.com/PostHog/posthog/issues/5363). Depending on the intended specificity of this endpoint, it could be nested under `/projects/(:id|@current)/` or  `/users/(:id|@me)/`.
This is where `StructuredViewSetMixin` will come in useful.
What does `StructuredViewSetMixin` _do_?
Well, it allows DRF `ViewSet`s to be nested, see endpoints nested under `/projects/` (`projects_router`) or `/organizations/` (`organizations_router`). Also, you're right that without `team_id` on `FeatureFlagOverride`, there's no way to not leak that across projects. The way this model is right now, it can only be nested under `/users/`.